processor/src/main/java/org/mobilitydata/gtfsvalidator/processor/TableLoaderGenerator.java

```diff
@@ -0,0 +1,218 @@
+        ArrayList<String> fieldNames = new ArrayList<>();
+        fieldNames.ensureCapacity(fileDescriptor.fields().size());
+        for (GtfsFieldDescriptor field : fileDescriptor.fields()) {
+            fieldNames.add(fieldNameField(field.name()));
+        }
+
+        MethodSpec.Builder method = MethodSpec.methodBuilder("getColumnNames")
+                .addAnnotation(Override.class)
+                .addModifiers(Modifier.PUBLIC)
+                .returns(ParameterizedTypeName.get(Set.class, String.class))
+                .addStatement("return $T.of($L)",
+                        ImmutableSet.class, String.join(", ", fieldNames));
+
+        return method.build();
+    }
+
+    private MethodSpec generateGetRequiredColumnNamesMethod() {
+        ArrayList<String> fieldNames = new ArrayList<>();
+        fieldNames.ensureCapacity(fileDescriptor.fields().size());
+        for (GtfsFieldDescriptor field : fileDescriptor.fields()) {
+            if (field.required()) {
+                fieldNames.add(fieldNameField(field.name()));
+            }
+        }
+
+        MethodSpec.Builder method = MethodSpec.methodBuilder("getRequiredColumnNames")
+                .addAnnotation(Override.class)
+                .addModifiers(Modifier.PUBLIC)
+                .returns(ParameterizedTypeName.get(Set.class, String.class))
+                .addStatement("return $T.of($L)",
+                        ImmutableSet.class, String.join(", ", fieldNames));
+
+        return method.build();
+    }
+
+    private MethodSpec generateLoadMethod() {
+        TypeName gtfsEntityType = classNames.entityImplementationTypeName();
+        TypeName tableContainerTypeName = classNames.tableContainerTypeName();
+        MethodSpec.Builder method = MethodSpec.methodBuilder("load")
+                .addModifiers(Modifier.PUBLIC)
+                .addParameter(Reader.class, "reader")
+                .addParameter(ValidatorLoader.class, "validatorLoader")
+                .addParameter(NoticeContainer.class, "noticeContainer")
+                .returns(ParameterizedTypeName.get(ClassName.get(GtfsTableContainer.class), gtfsEntityType))
+                .addStatement("$T csvFile = new $T(reader, FILENAME)", CsvFile.class, CsvFile.class)
+                .beginControlFlow("if (csvFile.isEmpty())")
+                .addStatement(
+                        "noticeContainer.addNotice(new $T(FILENAME))", EmptyFileNotice.class)
+                .addStatement(
+                        "$T table = $T.forEmptyFile()", tableContainerTypeName, tableContainerTypeName)
+                .addStatement(
+                        "validatorLoader.invokeSingleFileValidators(table, noticeContainer)")
+                .addStatement("return table")
+                .endControlFlow()
+                .beginControlFlow("if (!new $T().validate(FILENAME, csvFile.getColumnNames(), " +
+                                "getColumnNames(), getRequiredColumnNames(), noticeContainer))",
```

Comment:
If I understand correctly, csvFile.getColumnNames() returns column names as they are written in CSV file. getColumnNames and getRequireColumnNames both return something like LONG_NAME_FIELD_NAME. Am I missing anything?

Reply:
All these methods return lower_case_names.

```
public static final String LEVEL_ID_FIELD_NAME = "level_id";

  public static final String LEVEL_INDEX_FIELD_NAME = "level_index";

  public static final String LEVEL_NAME_FIELD_NAME = "level_name";
  @Override
  public Set<String> getColumnNames() {
    return ImmutableSet.of(LEVEL_ID_FIELD_NAME, LEVEL_INDEX_FIELD_NAME, LEVEL_NAME_FIELD_NAME);
  }

  @Override
  public Set<String> getRequiredColumnNames() {
    return ImmutableSet.of(LEVEL_ID_FIELD_NAME, LEVEL_INDEX_FIELD_NAME);
  }
```